docs/rfcs/1013-parsec-auth-scheme.md

```diff
@@ -0,0 +1,267 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+# Schema for parsec-auth
+
+## Overview
+
+This RFC defines the schema used for communicating with the parsec-auth service.
+
+## Goals and Non-Goals
+
+The goal is to define the required schema used when communicating with the parsec-auth service.
+For simplicity, we will skip the fido2 part that will be defined in a later RFC.
+
+## Design
+
+### Account creation
+
+To start the account creation, the client start by sending its email.
+
+```yml
+{
+  "cmd": "account_creation_step1_email_validation",
+  "req": {
+    "email": "Email"
+  },
+  "reps": [
+    {
+      "status": "ok"
+    },
+    {
+      "status": "invalid_email"
+    }
+  ]
+}
+```
+
+On `ok`, the server would have sent a mail with a unique token used for next the request used to register the authentication method:
+
+```yml
+{
+  "cmd": "account_creation_step2_registration",
+  "req": {
+    "email_validation_token": "EmailValidationToken",
+    "account_manifest": "EncryptedAccountManifest",
+    "auth_medium": "AuthMedium"
+  },
+  "reps": [
+    {
+      "status": "ok",
+      "auth_medium_id": "AuthMediumID"
+    },
+    {
+      "status": "invalid_email_validation_token"
+    }
+  ],
+  "custom_types": {
+    "AuthMedium": {
+      "id": "AuthMediumID",
+      "account_manifest_encrypted_sym_key": "EncryptedSymKey",
```

Comment:
We need a comment for each key field here explaining what they are needed for
(typically that `hmac_key` is provided in cleartext given it's a shared secret between the client and the server in order to do authenticate the client requests)

Reply:
done